### PR TITLE
fix: add custom height to images in Tabs

### PR DESCRIPTION
**Pull request by yuberdysheva**

### CODE REVIEW

File: src/blocks/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -37,15 +38,32 @@ export const TabsBlock = ({
     const isReverse = direction === 'content-media';
     const ref = useRef<HTMLDivElement>(null);
     const mediaWidth = ref?.current?.offsetWidth;
-    const mediaHeight = mediaWidth && getHeight(mediaWidth);
     const captionId = useUniqId();
-
+    const themedMedia = getThemedValue(activeTabData?.media, theme);
+    const mediaVideoHeight = mediaWidth && getHeight(mediaWidth);
+    const setVideoHeight = !themedMedia?.image && !activeTabData?.image;
```
Comment (gorgeousvlad) [inline src/blocks/Tabs/Tabs.tsx:44]: `const hasNoImage = !themedMedia?.image && !activeTabData?.image;`
`const mediaVideoHeight = hasNoImage && mediaWidth && getHeight(mediaWidth)`

File: src/blocks/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -91,15 +94,18 @@ export const TabsBlock = ({
             }}
             className={b('col', {centered: centered})}
         >
-            <div ref={ref}>
+            <div style={{minHeight: setVideoHeight ? mediaVideoHeight : minImageHeight}}>
```
Comment (gorgeousvlad) [inline src/blocks/Tabs/Tabs.tsx:97]: `<div style={{minHeight: mediaVideoHeight || minImageHeight}}>`

File: src/blocks/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -91,15 +94,18 @@ export const TabsBlock = ({
...
-                        playVideo={play}
-                        height={mediaHeight}
-                    />
+                    <div ref={ref}>
+                        <Media
+                            {...getThemedValue(activeTabData.media, theme)}
+                            key={activeTab}
+                            className={b('media')}
+                            playVideo={play}
+                            height={setVideoHeight ? mediaVideoHeight : undefined}
```
Comment (gorgeousvlad) [inline src/blocks/Tabs/Tabs.tsx:105]: `height={mediaVideoHeight || undefined}`

File: src/blocks/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -37,15 +38,32 @@ export const TabsBlock = ({
...
+    const themedMedia = getThemedValue(activeTabData?.media, theme);
+    const mediaVideoHeight = mediaWidth && getHeight(mediaWidth);
+    const setVideoHeight = !themedMedia?.image && !activeTabData?.image;
+    const [minImageHeight, setMinImageHeight] = useState(ref?.current?.offsetHeight);
+    // TODO remove property support activeTabData?.image. Use only activeTabData?.media?.image
     let imageProps;
 
+    const handleImageHeight = useCallback(() => {
+        setMinImageHeight(ref?.current?.offsetHeight);
+    }, []);
```
Comment (gorgeousvlad) [inline src/blocks/Tabs/Tabs.tsx:51]: should we add `ref?.current?.offsetHeight` to hook dependencies?

File: src/components/Media/Image/Image.tsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ export interface ImageAdditionProps {
     imageClassName?: string;
     isBackground?: boolean;
     fullscreen?: boolean;
+    handleImageLoad?: () => void;
```
Comment (gorgeousvlad) [inline src/components/Media/Image/Image.tsx:23]: `onLoad`

File: src/components/Media/Media.tsx [hunk 6]
```diff
@@ -43,6 +43,7 @@ export const Media = (props: MediaAllProps) => {
         qa,
         ratio,
         autoplay,
+        handleImageLoad,
```
Comment (gorgeousvlad) [inline src/components/Media/Media.tsx:46]: `onImageLoad`

File: src/components/Media/Media.tsx [hunk 7]
```diff
@@ -65,6 +66,7 @@ export const Media = (props: MediaAllProps) => {
                     hasVideoFallback={hasVideoFallback}
                     fullscreen={fullscreen}
                     qa={qaAttributes.image}
+                    handleImageLoad={handleImageLoad}
```
Comment (gorgeousvlad) [inline src/components/Media/Media.tsx:69]: `onLoad`

File: src/components/Media/Media.tsx [hunk 8]
```diff
@@ -104,6 +106,7 @@ export const Media = (props: MediaAllProps) => {
                     fullscreen={fullscreen}
                     analyticsEvents={analyticsEvents}
                     autoplay={autoplay}
+                    handleImageLoad={handleImageLoad}
```
Comment (gorgeousvlad) [inline src/components/Media/Media.tsx:109]: `onLoad`

File: src/components/VideoBlock/VideoBlock.tsx [hunk 9]
```diff
@@ -60,6 +60,7 @@ export interface VideoBlockProps extends AnalyticsEventsBase {
     height?: number;
     fullscreen?: boolean;
     autoplay?: boolean;
+    handleImageLoad?: () => void;
```
Comment (gorgeousvlad) [inline src/components/VideoBlock/VideoBlock.tsx:63]: `onImageLoad`

**Review by gorgeousvlad — CHANGES_REQUESTED** [review]

File: src/blocks/Tabs/TabsTextContent/TabsTextContent.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    showMedia,
+    data,
+    imageProps,
+    isReverse,
+}: TextContentProps) => (
+    <Col sizes={{all: 12, md: showMedia ? 4 : 8}} className={b({centered: centered})}>
+        <div
+            className={b('wrapper', {
+                reverse: isReverse,
+                regular: Boolean((data?.media || imageProps) && !isReverse),
```
[the remark this answers is not in the document]
Comment (yuberdysheva) [reply src/blocks/Tabs/TabsTextContent/TabsTextContent.tsx:32]: It's the opposite of reverse. Position of content: `media-text` | `text-media`

File: src/blocks/Tabs/Tabs.tsx (hunk 4 above)
Comment (yuberdysheva) [reply src/blocks/Tabs/Tabs.tsx:51]: ESLINT said that is unnecessary dependency

File: src/blocks/Tabs/TabsTextContent/TabsTextContent.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (gorgeousvlad) [reply src/blocks/Tabs/TabsTextContent/TabsTextContent.tsx]: could we have just default and reverse?

**Review by gorgeousvlad — APPROVED** [review]

**Pull request merged**